WIP: add variable examples (#4)
Garllon/mentorship: lessons/1-variables.md

## Lesson 1 - Variables - DataTypes

- [Variables](#variables)
  - [Number](#number)
  - [Text(String)](#textstring)
  - [List(Array)](#listarray)
  - [Boolean](#boolean)
- [DataTypes](#datatypes)
- [Examples](#examples)

## Variables

> Every practical programming language has a feature to do this, called variables.

Reviewer: What do you think about either omitting this sentence or rephrasing it to something like `Every practical programming language has a feature called variables.`

This is basically the same concept that you might know from math, although in
Ruby there are different kinds of variables (you will get to know another one in

> a couple chapters).

Reviewer: ```suggestion
a couple lessons).
```

In Ruby you can assign a name to something (an object) by using the so called
assignment operator =. On the left side you have the variable name and on the
right side, what you like to assign.

### Number

Define an number.
```ruby
# An number
number = 1
```

Calculcate with numbers
```ruby
number1 = 10
number2 = 5

# Use the + operator
result = number1 + number2
puts result 
# => 15

# Use the / operator
result = number1 / number2
puts result 
# => 0
```

> Why is the last result 0 instead of 0.5 what we would expect?

Reviewer: ```suggestion
Why is the last result 0 instead of 0.5 as expected?
```

> Integer is limited to whole numbers. If we like to do this operation correct
> we need floating point numbers. So the same operation as before, except that
> we define the numbers as floats.

Reviewer: ```suggestion
Integer is limited to whole numbers. If we want to do this operation correctly then
we need floating point numbers. So the same operation as before, except this time
we define the numbers as floats.
```

```ruby
number1 = 10.0
number2 = 5.0

# Use the / operator
result = number1 / number2
puts result 
# => 0.5
```

> Why not always use floats is maybe a question you have? Floats needs more
> memory and in the computer science you like to optimize your code.
> Think power usage. Your smartphone live longer, before you need to recharge. :)

Reviewer: ```suggestion
Why not always use floats you ask? Floats need more
resources and we like to optimize code.
Think power usage. Your smartphone can live longer before you need to recharge. :)
```
